docs/api/tensorflow_usage.rst

```diff
@@ -14,19 +14,32 @@ All configuration options are specified using environment variables as shown in
 Optimizing CPU Performance
 --------------------------
 
+.. note::
+    It is advisable to configure these options only if you are an advanced tensorflow user and understand
+    how different operations are implemented under the hood.
+
+These options are useful if you have implemented a custom component as part of your pipeline. Based on the tensorflow operations
+that the component uses, you can leverage multi-core CPU parallelism by tuning these options.
+
 Parallelizing One Operation
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Set ``TF_INTRA_OP_PARALLELISM_THREADS`` as an environment variable to specify the maximum number of threads that can be used
-to parallelize the execution of one operation. If left unspecified, this value defaults to ``0`` which means TensorFlow should
+to parallelize the execution of one operation. For example, operations like ``tf.matmul()`` and ``tf.reduce_sum`` can be executed
+on multiple threads running parallely. The default value for this variable is ``0`` which means TensorFlow should
 pick an appropriate value depending on the system configuration.
```

Comment:
what does "an appropriate value" actually mean? That it'll run on the maximum amount of possible threads? If yes, why even set this variable to anything?

Reply:
@akelad TensorFlow also does not mention anything about it in the docs. Assuming it runs on maximum amount of possible threads, as a user this may be undesirable because that hogs my CPU compute capacity and I may want to be conservative in it for other processed.

Reply:
Okay, I looked at the implementation and default value of 0 makes tensorflow allocate 1 thread per CPU core. Updated this here.